src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -51,101 +51,58 @@ class RadialBasisFctMapping : public Mapping {
-
-  virtual void tagMeshSecondRound() override;
-
 private:
   precice::logging::Logger _log{"mapping::RadialBasisFctMapping"};
 
-  bool _hasComputedMapping = false;
-
-  /// Radial basis function type used in interpolation.
-  RADIAL_BASIS_FUNCTION_T _basisFunction;
-
   Eigen::MatrixXd _matrixA;
 
   Eigen::ColPivHouseholderQR<Eigen::MatrixXd> _qr;
 
-  /// true if the mapping along some axis should be ignored
-  std::vector<bool> _deadAxis;
-
-  void mapConservative(int inputDataID, int outputDataID, int polyparams);
-  void mapConsistent(int inputDataID, int outputDataID, int polyparams);
-
-  void setDeadAxis(bool xDead, bool yDead, bool zDead)
-  {
-    _deadAxis.resize(getDimensions());
-    if (getDimensions() == 2) {
-      _deadAxis[0] = xDead;
-      _deadAxis[1] = yDead;
-      PRECICE_CHECK(not(xDead && yDead), "You cannot choose all axes to be dead for a RBF mapping");
-      if (zDead)
-        PRECICE_WARN("Setting the z-axis to dead on a 2-dimensional problem has no effect.");
-    } else if (getDimensions() == 3) {
-      _deadAxis[0] = xDead;
-      _deadAxis[1] = yDead;
-      _deadAxis[2] = zDead;
-      PRECICE_CHECK(not(xDead && yDead && zDead), "You cannot choose all axes to be dead for a RBF mapping");
-    } else {
-      PRECICE_ASSERT(false);
-    }
-  }
+  /// @copydoc RadialBasisFctBaseMapping::mapConservative
+  virtual void mapConservative(DataID inputDataID, DataID outputDataID) override;
+
+  /// @copydoc RadialBasisFctBaseMapping::mapConsistent
+  virtual void mapConsistent(DataID inputDataID, DataID outputDataID) override;
 };
 
 // --------------------------------------------------- HEADER IMPLEMENTATIONS
 
 template <typename RADIAL_BASIS_FUNCTION_T>
 RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::RadialBasisFctMapping(
-    Constraint              constraint,
+    Mapping::Constraint     constraint,
     int                     dimensions,
     RADIAL_BASIS_FUNCTION_T function,
     bool                    xDead,
     bool                    yDead,
     bool                    zDead)
-    : Mapping(constraint, dimensions),
-      _basisFunction(function)
+    : RadialBasisFctBaseMapping<RADIAL_BASIS_FUNCTION_T>(constraint, dimensions, function, xDead, yDead, zDead)
 {
-  if (constraint == SCALEDCONSISTENT) {
-    setInputRequirement(Mapping::MeshRequirement::FULL);
-    setOutputRequirement(Mapping::MeshRequirement::FULL);
-  } else {
-    setInputRequirement(Mapping::MeshRequirement::VERTEX);
-    setOutputRequirement(Mapping::MeshRequirement::VERTEX);
-  }
-  setDeadAxis(xDead, yDead, zDead);
 }
 
 template <typename RADIAL_BASIS_FUNCTION_T>
 void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
 {
   PRECICE_TRACE();
 
-  precice::utils::Event e("map.rbf.computeMapping.From" + input()->getName() + "To" + output()->getName(), precice::syncMode);
+  precice::utils::Event e("map.rbf.computeMapping.From" + this->input()->getName() + "To" + this->output()->getName(), precice::syncMode);
```

Comment:
What is the added value of adding "this" ?

Reply:
The compiler forces me to explicitly refer to the base class member variables here, as the base class is now a class template with a template parameter depending on the instantiation of the derived class. If not put here, the compiler will not find the variable in the first place.